elasticsearch/templates/deployment-es-master.yaml

```diff
@@ -0,0 +1,71 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: es-master
+  labels:
+    component: {{ .Values.labels.es_master.component }}
+    role: {{ .Values.labels.es_master.role }}
+spec:
+  replicas: {{ .Values.replicas.es_master }}
+  template:
+    metadata:
+      labels:
+        component: {{ .Values.labels.es_master.component }}
+        role: {{ .Values.labels.es_master.role }}
+      annotations:
+        pod.beta.kubernetes.io/init-containers: '[
+          {
+          "name": "sysctl",
```

Comment:
This init container isn't necessary.  Elasticsearch has two methods for changing heap size:  the first is via the ES_HEAP_SIZE environment variable, and the second is with JVM flags.  The example used here (ES_JAVA_OPTS) sets these for the container already.  We should look to see what the contents of the elasticsearch configuration file include to provide a method to override the default configuration values through a templated config file in a confmap, or we should make sure the environment variables used in the current approach can be customized through values.yaml